x-pack/legacy/plugins/siem/server/lib/detection_engine/notifications/rules_notification_alert_type.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger } from 'src/core/server';
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { NOTIFICATIONS_ID } from '../../../../common/constants';
+
+import { NotificationAlertTypeDefinition } from './types';
+import { getSignalsCount } from './get_signals_count';
+import { RuleAlertAttributes } from '../signals/types';
+import { scheduleNotificationActions } from './schedule_notification_actions';
+
+export const rulesNotificationAlertType = ({
+  logger,
+  kibanaUrl,
+}: {
+  logger: Logger;
+  kibanaUrl: string;
+}): NotificationAlertTypeDefinition => ({
+  id: NOTIFICATIONS_ID,
+  name: 'SIEM Notifications',
+  actionGroups: [
+    {
+      id: 'default',
+      name: i18n.translate(
+        'xpack.siem.detectionEngine.ruleNotificationAlert.actionGroups.default',
+        {
+          defaultMessage: 'Default',
+        }
+      ),
+    },
+  ],
+  defaultActionGroupId: 'default',
+  validate: {
+    params: schema.object({
+      ruleAlertId: schema.string(),
+    }),
+  },
+  async executor({ startedAt, previousStartedAt, alertId, services, params }) {
+    const ruleAlertSavedObject = await services.savedObjectsClient.get<RuleAlertAttributes>(
+      'alert',
+      params.ruleAlertId
+    );
+
+    if (!ruleAlertSavedObject.attributes.params) {
+      logger.error(`Saved object for alert ${params.ruleAlertId} was not found`);
+      return;
+    }
+
+    const { params: ruleAlertParams, name: ruleName } = ruleAlertSavedObject.attributes;
+    const ruleParams = { ...ruleAlertParams, name: ruleName };
+
+    const { signalsCount, resultsLink } = await getSignalsCount({
+      from: previousStartedAt ?? `now-${ruleParams.interval}`,
+      to: startedAt,
```

Comment:
Not sure if this daterange is the issue or if it's timing from the signals being written, but I'm sometimes seeing mis-match signals counts in the server logs between `signalsRulesAlertType` and `rulesNotificationsAlertType`, and then the action is not fired.

e.g. here's the server log from when a single rule runs:

```
server    log   [17:00:04.302] [info][plugins][siem] Found 55 signals from the indexes of "[apm-*-transaction*, auditbeat-*, endgame-*, filebeat-*, packetbeat-*, winlogbeat-*]" using signal rule name: "Signal Generator 7k", id: "c0880b7e-0bbb-4ef2-9471-056d36741ba0", rule_id: "38bcddb9-8751-40b8-8244-04254c7a818c", pushing signals to index ".siem-signals-spong-default"
server    log   [17:00:05.267] [info][plugins][siem] Found 0 signals using signal rule name: "Signal Generator 7k", id: "38bcddb9-8751-40b8-8244-04254c7a818c", rule_id: "38bcddb9-8751-40b8-8244-04254c7a818c" in ".siem-signals-spong-default" index
```

Which **does not** result in the rule's action being completed (slack message in this instance). No explicit repro steps yet, but I've noticed it mostly when editing an action after a rule has been created, and then this happens the next time the rule runs.